### PR TITLE
Radius Audit Log

### PR DESCRIPTION
**Pull request by jrouzierinverse**
## Description

Radius audit log 
## Impacts

RADIUS authorize workflow
## NEWS file entries

New Features
++++++++++++
- Radius audit log
## Delete branch after merge

NO

### CODE REVIEW

File: conf/pf.conf.defaults [hunk 1]
```diff
@@ -408,6 +408,14 @@ traplog=0D
...
 # locationlog=180D
 locationlog=0D
+#
+# expire.radius_audit_log
+#
+# Time which you would like to keep radius audit log entries
+# A value of 0D disables expiration.
+# example:
+# radius_audit_log=180D
+radius_audit_log=0D
```
Comment (fdurand) [inline conf/pf.conf.defaults]: We must be agressive with the default value, let's say a week.

File: lib/pf/radius.pm [hunk 2]
```diff
@@ -60,7 +60,7 @@ sub new {
     my $logger = get_logger();
     $logger->debug("instantiating new pf::radius object");
     my ( $class, %argv ) = @_;
-    my $self = bless {}, $class;
+    my $self = bless { stash => {} }, $class;
```
Comment (fdurand) [inline lib/pf/radius.pm]: do we have to keep the stash ?

File: db/radius_audit_log.sql [hunk 3]
```diff
@@ -0,0 +1,50 @@
+--
+-- Table structure for table 'radius_audit_log'
+--
+
+DROP TABLE IF EXISTS radius_audit_log;
+
+CREATE TABLE radius_audit_log (
```
Comment (fdurand) [inline db/radius_audit_log.sql:7]: Must be move to pf-schema and upgrade-schema

File: addons/extract_i18n_strings.pl [hunk 4]
```diff
@@ -318,6 +319,7 @@ sub extract_modules {
     const('pf::config', 'SoH Status', \@pf::config::SOH_STATUS);
     const('pf::config', 'Inline triggers', [$pf::config::MAC, $pf::config::PORT, $pf::config::SSID, $pf::config::ALWAYS]);
     const('pf::config', 'Network types', [$pf::config::NET_TYPE_VLAN_REG, $pf::config::NET_TYPE_VLAN_ISOL, $pf::config::NET_TYPE_INLINE, 'management', 'other']);
+    const('pf::radius_audit_log', 'Radius Aduit Log', \@pf::radius_audit_log::FIELDS);
```
Comment (julsemaan) [inline addons/extract_i18n_strings.pl]: Radius Aduit Log -> RADIUS Audit Log

**Comment by jrouzierinverse** [comment]
This has been fixed

File: conf/pf.conf.defaults [hunk 5]
```diff
@@ -408,6 +408,14 @@ traplog=0D
...
 # locationlog=180D
 locationlog=0D
+#
+# expire.radius_audit_log
+#
+# Time which you would like to keep radius audit log entries
+# A value of 0D disables expiration.
+# example:
+# radius_audit_log=180D
+radius_audit_log=1W
```
Comment (louismunro) [inline conf/pf.conf.defaults]: I humbly suggest 30D as a default.
I think most clients will want more than 1W.

File: addons/database-backup-and-maintenance.sh [hunk 6]
```diff
@@ -70,6 +70,8 @@ if [ -f /var/run/mysqld/mysqld.pid ]; then
     
     /usr/local/pf/addons/database-cleaner.pl --table=auth_log --date-field=attempted_at --older-than="1 WEEK"
 
+    /usr/local/pf/addons/database-cleaner.pl --table=radius_audit_log --date-field=created_at --older-than="1 WEEK"
```
Comment (fdurand) [inline addons/database-backup-and-maintenance.sh]: probably have to change to 30D too.

**Pull request merged**